jwst/guider_cds/guider_cds.py

```diff
@@ -136,119 +120,52 @@ def guider_cds(model, gain_model=None, readnoise_model=None):
-        readnoise_arr = np.zeros(imshape, dtype=np.float32) + DEFAULT_READNOISE
-
-        log.info('A READNOISE reference file could not be retrieved from CRDS.')
-        log.info('The mean of the SCI values in the FGS readnoise reference file %s', DEFAULT_READNOISE_FILE)
-        log.info('will be used to populate the readnoise array.')
+    # extract subarray from reference file, if necessary
+    if reffile_utils.ref_matches_sci(model, reference_model):
+        ref_arr = reference_model.data
     else:
-        # extract subarray from readnoise reference file, if necessary
-        if reffile_utils.ref_matches_sci(model, readnoise_model):
-            readnoise_arr = readnoise_model.data
-        else:
-            log.info('Extracting readnoise reference file subarray to match science data')
-            ref_sub_model = reffile_utils.get_subarray_model(model, readnoise_model)
-            readnoise_arr = ref_sub_model.data
+        try:
+            ref_sub_model = reffile_utils.get_subarray_model(model, reference_model)
+            log.info(
+                f"Extracting subarray from reference {reference_model.meta.model_type} "
+                f"to match science data."
+            )
+            ref_arr = ref_sub_model.data
             ref_sub_model.close()
-
-    return gain_arr, readnoise_arr
-
-
-def get_dataset_info(model):
-    """
-    Short Summary
-    -------------
-    Extract values for the image shape, the number of integrations,
-    the group time, and the exposure type.
-
-    Parameters
-    ----------
-    model: instance of Data Model
-       DM object for input
-
-    Returns
-    -------
-    imshape: (int, int) tuple
-       shape of 2D image
-
-    n_int: int
-       number of integrations
-
-    grp_time: float
-       group time
-
-    exp_type: string
-        exposure type
-    """
-    instrume = model.meta.instrument.name
-    frame_time = model.meta.exposure.frame_time
-    ngroups = model.meta.exposure.ngroups
-    grp_time = model.meta.exposure.group_time
-    exp_type = model.meta.exposure.type
-
-    n_int = model.data.shape[0]
-    asize2 = model.data.shape[2]
-    asize1 = model.data.shape[3]
-
-    imshape = (asize2, asize1)
-
-    log.info('Instrument: %s' % (instrume))
-    log.info('Exposure type: %s' % (exp_type))
-    log.info('Number of integrations: %d' % (n_int))
-    log.info('Number of groups per integration: %d' % (ngroups))
-    log.info('Group time: %s' % (grp_time))
-    log.info('Frame time: %10.5f' % (frame_time))
-
-    return imshape, n_int, grp_time, exp_type
+        except ValueError as err:
+            if "STACK" in model.meta.exposure.type:
+                if model.data.shape[-2:] != reference_model.data.shape:
+                    log.debug(
+                        f"The {reference_model.meta.model_type} reference file array does not "
+                        f"match the shape of stacked FGS data;"
+                        f"applying the mean value to the data."
```

Comment:
Only the first one needs to be an f-string.  It's NBD either way, and I don't really care if you leave it as-is, but I'm surprised the ruff rules didn't notice this